pkg/apis/triggers/v1alpha1/event_listener_defaults.go

```diff
@@ -5,4 +5,24 @@ import (
 )
 
 // SetDefaults sets the defaults on the object.
-func (el *EventListener) SetDefaults(ctx context.Context) {}
+func (el *EventListener) SetDefaults(ctx context.Context) {
+	if IsUpgradeViaDefaulting(ctx) {
+		// Most likely those EventListener passed here is already running
+		for i := range el.Spec.Triggers {
+			t := &el.Spec.Triggers[i]
+			if t.DeprecatedBinding != nil {
+				if len(t.Bindings) > 0 {
+					// Both binding & bindings are set...we should validate for this?
+					// Maybe we should check to see if deprecatedBinding is present in Bindings?
```

Comment:
Yeah. I think should make sure this cannot happen via validation.
For old resources, `bindings` was not defined, so will not be there. 
For new resources, if they try to define both `bindings` and `binding`, they should not be accepted.

Reply:
Added validation for this!